src/garage/tf/embeddings/gaussian_mlp_encoder.py

```diff
@@ -150,15 +166,20 @@ def dist_info_sym(self, input_var, state_info_vars=None, name='default'):
                                                            name=name)
         return dict(mean=mean_var, log_std=log_std_var)
 
+    @property
+    def spec(self):
+        """garage.InOutSpec: Specification of input and output."""
+        return self._embedding_spec
+
     @property
     def input_dim(self):
         """int: Dimension of the encoder input."""
-        return self._embedding_spec.input_space.flat_dim
+        return self.spec.input_space.flat_dim
```

Comment:
FYI you don't need to refer to properties in their implemena

Reply:
OK. I felt that referring this property makes the line shorter

Reply:
you run the risk of executing a function call (a property is a function) rather than just referencing a variable, but this is not really a performance issue for a method like this.